projects/extension/src/background/ConnectionManager.ts

```diff
@@ -386,6 +381,10 @@ export class ConnectionManager
         return this.#handleSpecMessage(msg, app)
       }
 
+      if (!msg.payload) {
+        return
```

Comment:
shouldn't this have the same treatment as the other unrecognized messages?

Reply:
msg is already checked on the flow and is valid type-wise. If payload is empty then there is no reason to throw an error,  add it to the `pendingRequests` or propagate it to `healthChecker`.